src/redux/actions/samples/updateSampleFileUploadV2.js

```diff
@@ -0,0 +1,56 @@
+import moment from 'moment';
+
+import { SAMPLES_FILE_UPDATE } from 'redux/actionTypes/samples';
+import endUserMessages from 'utils/endUserMessages';
+import fetchAPI from 'utils/http/fetchAPI';
+import handleError from 'utils/http/handleError';
+
+const fileNameForApiV1 = {
+  matrix10x: 'matrix.mtx.gz',
+  barcodes10x: 'barcodes.tsv.gz',
+  features10x: 'features.tsv.gz',
+};
+
+const updateSampleFileUploadV2 = (
+  experimentId, sampleId, type, uploadStatus, uploadProgress,
+) => async (dispatch) => {
+  // Don't send an api update whenever the progress bar is updated, only for uploadStatus changes
+  // TODO: move progress to not even be a part of redux, manage it in a different way
+  if (!uploadProgress) {
+    const url = `/v2/experiments/${experimentId}/samples/${sampleId}/sampleFiles/${type}`;
+    const body = { uploadStatus };
+
+    try {
+      await fetchAPI(
+        url,
+        {
+          method: 'PATCH',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(body),
+        },
+      );
+    } catch (e) {
+      // This error means that we couldn't update the api on the state of the upload
+      // Should we block the upload from continuing? The upload itself isn't necessarily compromised
+      handleError(e, endUserMessages.ERROR_UPDATE_SERVER_ON_UPLOAD_STATE);
+    }
+  }
+
+  const updatedAt = moment().toISOString();
+
+  // Even if we failed to persist the update in the api it is still
+  // valid information so we should dispatch the action anyways
```

Comment:
Wouldn't this cause inconsistent state if the user reloads the page?

Reply:
I didn't like  the alernatives I could come up with. It would also be inconsistent if we don't say anything about the upload continuing even thought it did, right?

Do you have any ideas?

This is related to this other comment I wrote in that same action, what would you suggest?
> // This error means that we couldn't update the api on the state of the upload
> // Should we block the upload from continuing? The upload itself isn't necessarily compromised

Reply:
// This error means that we couldn't update the api on the state of the upload
// Should we block the upload from continuing? The upload itself isn't necessarily compromised

For this case, we depend on the files being uploaded and the upload status to be set to UPLOADED for users to be able to continue to process the data right? So if there are any erros updating the file, it might be just better to set the upload status to failed and ask the user to reupload, as there are no other checks to check whether the file has successfully uplaoded or not.

Reply:
I guess that follows for this comment as well

```
  // Even if we failed to persist the update in the api it is still
  // valid information so we should dispatch the action anyways
```

If we let the update continue, the user will be able to process the data on the first run because the all the sample files are uploaded. However, when they login again or come back to `data-managment`, the file upload status will not be UPLOADED, and they can't go to data processing. They'd have to reupload and process data again, right? So I think it's better to mark the upload as failed and let the user reupload rather than having them come back and see to see that they have to reprocess the data. Or, upon failure of saving, is it possible to retry several times?

Reply:
>  it might be just better to set the upload status to failed and ask the user to reupload, as there are no other checks to check whether the file has been successfully uploaded or not.

True

Reply:
good point, I'll just set to upload failed